### DISCUSSION TITLE
How Many Users can Jamulus Server Running on a Raspberry Pi Support?

### DISCUSSION THREAD

**Discussion by GLardner** [jamulussoftware/jamulus#2005] [General]
How many users can be hosted on a Jamulus server running on a Raspberry Pi 4? Assuming the broadband feed is not a limiting factor.

I am in a choir that has historically had 80-100 active members. Now we rehearse using Zoom and Jamulus. A core group use Jamulus, a feed from which is then fed back into the Zoom rehearsal. Last season we typically had 20-30 on Jamulus; we're hoping to encourage more of our members to use Jamulus this season. 

I don't have a good broadband feed at my home so I am contemplating setting up a Jamulus server on an RPi and 'hosting' it with another member who does have a better broadband connection. So the question is, would Jamulus on an RPi 4 be able to host 30 singers? Or 40 singers? Or more? How many users have you people here actually managed to host on a Raspberry Pi 4 successfully and without dropout or crackling?

**Comment by gene96817** [comment]
I successfully held weekly rehearsals on a Raspberry Pi 4 with 12-14 singers. I can't comment on the experience for others.

**Comment by mcfnord** [comment]
I don't know the actual answer. I hate doubting the hearty RasPi 4, but I suggest you consider a cloud-based server. A RasPi 4 can handle 5, and I'm sure I've seen ten, but 30 people on RasPi 4 doesn't sound wise to me. Both the hardware and the network inside a data center is built differently than a home internet connection to a RasPi 4. Fewer bottlenecks. 30 people's time is on the line, it deserves more than a home grade platform.

**Reply by gene96817** [reply]
How much does it cost to run Jamulus in a cloud-based server? I find it frustrating that no one has been willing to share what it cost them. I have heard it is very inexpensive. I have heard is  much more than beer money. I know  exactly what it cost to own and run a RasPi4.  When I needed to host more than 14 singers, I chose to buy a more powerful hardware. I know what that cost to own for the next year. No way I would go to the cloud without knowing if I can afford it. If you recommend using a cloud-baSed server, please give examples of real numerical cost. Thanks

**Reply by chrisrimple** [reply]
See the Jamulus section of this doc...

https://docs.google.com/document/d/1smcvsxdaaViPQvGMQHmah_6BQeqowhmGSFMHfnlY2FI/

**Reply by rdica** [reply]
@gene96817 I pay ~US$20 month for a VPS on Ramnode NY. I had this before I even knew about Jamulus, and now use this VPS for my server. It has 4 cpus, 4GB RAM, 100GB disk, and 4TB data xfer/month.

Other cloud service offerings are typically US$5-$20/month depending on cpu/mem/disk/data xfer specs.

Does this help you?

**Reply by chrisrimple** [reply]
I pay $6/month on Vultr.com for a server that's plenty powerful enough for my band's rehearsals (there are 6 of us).

**Reply by gene96817** [reply]
THANK YOU!

**Reply by gene96817** [reply]
@chrisrimple  Thank you. Sorry I didn't read your doc closely; it is quite comprehensive. I am a bit buried for the next two weeks. $6/mo flat rate is a great deal. I'll look at the fine print. I wonder what tier service would be needed for 30 singers.

**Comment by gene96817** [comment]
@rdica @chrisrimple   Thank you for the numbers.   @chrisrimple how many rehearsals a month do you have for the $6/month? When I was using a RasPi, I was hosting two 6-person  rehearsals and one 14-person rehearsal a week. If that cost me $12/month,  that could be less than using a cloud service. I bought a bigger piece of hardware because my chorus outgrew the RasPi4. 

I wonder  what it would cost to host a 20 or 30 person session once a week? If the cost of the cloud hosting is sensitive to the number of musician-minutes, I could easily be running at a $50/mo equivalent. I know my server does have visitors, too. I am definitely needing more than beer money. The nice thing about having  a fixed cost, I don't worry about having more  users.  

Philosophically, I like the idea of operating public servers as a community resource. I wouldn't be doing it if I have to worry about my monthly bill.

**Reply by chrisrimple** [reply]
If you check my doc (linked above), I provide details on the installation that I use.  Vultr.com is $6/month for unlimited use (for the configuration I've chosen), which is certainly powerful enough for ~15 musicians.  Above that, you may want to upgrade to the next tier, which might be $12/month.  My server is up 24/7 and my band is rehearsing 2x/week.

**Comment by mikedickey** [comment]
Fwiw you can get free, fully managed, on demand Jamulus servers from [Jacktrip Virtual Studio](https://app.jacktrip.com), through the end of the year.

**Reply by GLardner** [reply]
Thank you, everyone, for the advice and suggestions.
